internal/runners/packages/searchTable.go

```diff
@@ -0,0 +1,222 @@
+	keyWebsite     = locale.Tl("search_website", "Website")
+	keyLicense     = locale.Tl("search_license", "License")
+	keyVersions    = locale.Tl("search_versions", "Versions")
+	keyVulns       = locale.Tl("search_vulnerabilities", "Vulnerabilities (CVEs)")
+
+	keys = []string{
+		keyName,
+		keyDescription,
+		keyWebsite,
+		keyLicense,
+		keyVersions,
+		keyVulns,
+	}
+)
+
+type structuredSearchResults struct {
+	Results      []*searchResult `json:"Results,omitempty"`
+	packageNames []string
+	width        int
+	height       int
+}
+
+type searchResult struct {
+	Name            string         `json:"Name,omitempty"`
+	Description     string         `json:"Description,omitempty"`
+	Website         string         `json:"Website,omitempty"`
+	License         string         `json:"License,omitempty"`
+	Versions        []string       `json:"Versions,omitempty"`
+	Vulnerabilities map[string]int `json:"Vulnerabilities,omitempty"`
+	version         string
+}
+
+func createSearchTable(width, height int, packages []*model.IngredientAndVersion, vulns map[string][]*model.VulnerabilityIngredient) (*structuredSearchResults, error) {
+	maxKeyLength := 0
+	for _, key := range keys {
+		renderedKey := styleBold.Render(key)
+		if len(renderedKey) > maxKeyLength {
+			maxKeyLength = len(renderedKey) + 2
+		}
+	}
+
+	var results []*searchResult
+	var packageNames []string
+	for _, pkg := range packages {
+		result := &searchResult{}
+		if pkg.Ingredient.Name != nil {
+			result.Name = *pkg.Ingredient.Name
+		}
+		if pkg.Ingredient.Description != nil {
+			result.Description = *pkg.Ingredient.Description
+		}
+		if pkg.Ingredient.Website != "" {
+			result.Website = pkg.Ingredient.Website.String()
+		}
+		if pkg.LatestVersion.LicenseExpression != nil {
+			result.License = *pkg.LatestVersion.LicenseExpression
+		}
+
+		var versions []string
+		for i, v := range pkg.Versions {
+			if i > 5 {
+				versions = append(versions, fmt.Sprintf("... (%d more)", len(pkg.Versions)-5))
+				break
+			}
+			versions = append(versions, styleCyan.Render(v.Version))
+		}
+		if len(versions) > 0 {
+			result.Versions = versions
+		}
+		result.version = pkg.Version
+
+		ingredientVulns := vulns[ingredientVulnKey(*pkg.Ingredient.PrimaryNamespace, *pkg.Ingredient.Name, pkg.Version)]
+		if len(ingredientVulns) > 0 {
+			var (
+				critical int
+				high     int
+				medium   int
+				low      int
+			)
+			for _, v := range ingredientVulns {
+				critical += len(v.Vulnerabilities.Critical)
+				high += len(v.Vulnerabilities.High)
+				medium += len(v.Vulnerabilities.Medium)
+				low += len(v.Vulnerabilities.Low)
+			}
+
+			result.Vulnerabilities = make(map[string]int)
+			if critical > 0 {
+				result.Vulnerabilities["Critical"] = critical
+			}
+			if high > 0 {
+				result.Vulnerabilities["High"] = high
+			}
+			if medium > 0 {
+				result.Vulnerabilities["Medium"] = medium
+			}
+			if low > 0 {
+				result.Vulnerabilities["Low"] = low
+			}
+		}
```

Comment:
This feels like it should live in the model, maybe where `vulns[<key>]` is being calculated.